### DISCUSSION TITLE
Set `networkMode` to 'always' if `NODE_ENV` is 'development'?

### DISCUSSION THREAD

**Discussion by iyxan23** [TanStack/query#8019] [Ideas]
Hello! Thanks for your awesome library, I'm coming from tRPC land and was confused as to why there are no requests happening as I open up my NextJS local dev server. I took a look at `react-query` and it appears that it only fetches when the device is online (and it just so happens that I'm using my laptop offline), which does make sense as a default.

I did a simple configuration change on my `QueryClient` as such:

```diff
-const createQueryClient = () => new QueryClient();
+const createQueryClient = () =>
+  new QueryClient({
+    defaultOptions: {
+      queries: {
+        networkMode: env.NODE_ENV === "development" ? "always" : "online",
+      },
+      mutations: {
+        networkMode: env.NODE_ENV === "development" ? "always" : "online",
+      },
+    },
+  });
```

That made me wonder, the default of `"online"` would make a bit more sense when `NODE_ENV` is `development`. I don't think developers will need to assure that they are online or not to test their web apps.

And so I propose that the default `networkMode` for `react-query` be set to "always" when `NODE_ENV` is `development`. Just a small suggestion from me, thank you!

**Comment by TkDodo** [comment]
I don't think it's ideal to have different settings in dev vs prod mode, as it might surprise users then. With this default, they will never see a `fetchStatus: 'paused'` and production might be the first time that they encounter this. It also makes the offline toggle button in the query devtools useless.

So, it's relatively easy to set the desired defaults yourself, as you've shown, but we won't do this in the lib internally.

**Reply by iyxan23** [reply]
Ah right, thank you! :)
